akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java

```diff
@@ -430,7 +430,24 @@ public void apply(String elem) {
     probe.expectNoMsg(FiniteDuration.create(200, TimeUnit.MILLISECONDS));
     probe.expectMsgEquals("tick");
     probe.expectNoMsg(FiniteDuration.create(200, TimeUnit.MILLISECONDS));
+  }
 
+  @Test
+  public void mustProduceTicksWithJavaDuration() throws Exception {
+    final TestKit probe = new TestKit(system);
+    Source<String, Cancellable> tickSource = Source.tick(java.time.Duration.ofSeconds(1),
+            java.time.Duration.ofMillis(500), "tick");
+    @SuppressWarnings("unused")
+    Cancellable cancellable = tickSource.to(Sink.foreach(new Procedure<String>() {
+      public void apply(String elem) {
+        probe.getRef().tell(elem, ActorRef.noSender());
+      }
+    })).run(materializer);
+    probe.expectNoMsg(FiniteDuration.create(600, TimeUnit.MILLISECONDS));
+    probe.expectMsgEquals("tick");
+    probe.expectNoMsg(FiniteDuration.create(200, TimeUnit.MILLISECONDS));
+    probe.expectMsgEquals("tick");
+    probe.expectNoMsg(FiniteDuration.create(200, TimeUnit.MILLISECONDS));
```

Comment:
Since the method does not really reimplement how ticking is done, I wonder if a test taking a full second is really worth it (remember we have a very extensive test suite and every slow test counts). Maybe just have a compile-only method making sure the method exists?

Reply:
Yeah make sense, there will be dozens of methods like this in the future.

Reply:
Done, renamed to `mustCompileMethodsWithJavaDuration` for future PRs